miio/airqualitymonitor_miot.py

```diff
@@ -0,0 +1,286 @@
+import enum
+import logging
+
+import click
+
+from .click_common import command, format_output
+from .exceptions import DeviceException
+from .miot_device import MiotDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+MODEL_AIRQUALITYMONITOR_CGDN1 = "cgllc.airm.cgdn1"
+
+_MAPPING_CGDN1 = {
+    # Source https://miot-spec.org/miot-spec-v2/instance?type=urn:miot-spec-v2:device:air-monitor:0000A008:cgllc-cgdn1:1
+    # Environment
+    "humidity": {"siid": 3, "piid": 1},  # [0, 100] step 1
+    "pm25": {"siid": 3, "piid": 4},  # [0, 1000] step 1
+    "pm10": {"siid": 3, "piid": 5},  # [0, 1000] step 1
+    "temperature": {"siid": 3, "piid": 7},  # [-30, 100] step 0.00001
+    "co2": {"siid": 3, "piid": 8},  # [0, 9999] step 1
+    # Battery
+    "battery": {"siid": 4, "piid": 1},  # [0, 100] step 1
+    "charging_state": {
+        "siid": 4,
+        "piid": 2,
+    },  # 1 - Charging, 2 - Not charging, 3 - Not chargeable
+    "voltage": {"siid": 4, "piid": 3},  # [0, 65535] step 1
+    # Settings
+    "start_time": {"siid": 9, "piid": 2},  # [0, 2147483647] step 1
+    "end_time": {"siid": 9, "piid": 3},  # [0, 2147483647] step 1
+    "monitoring_frequency": {
+        "siid": 9,
+        "piid": 4,
+    },  # 1, 60, 300, 600, 0; device accepts [0..600]
+    "screen_off": {
+        "siid": 9,
+        "piid": 5,
+    },  # 15, 30, 60, 300, 0; device accepts [0..300], 0 means never
+    "device_off": {
+        "siid": 9,
+        "piid": 6,
+    },  # 15, 30, 60, 0; device accepts [0..60], 0 means never
+    "temperature_unit": {"siid": 9, "piid": 7},
+}
+
+
+class AirQualityMonitorMiotException(DeviceException):
+    pass
+
+
+class ChargingState(enum.Enum):
+    Unplugged = 0  # Not mentioned in the spec
+    Charging = 1
+    NotCharging = 2
+    NotChargable = 3
+
+
+class MonitoringFrequencyCGDN1(enum.Enum):  # Official spec options
+    Every1Second = 1
+    Every1Minute = 60
+    Every5Minutes = 300
+    Every10Minutes = 600
+    NotSet = 0
+
+
+class ScreenOffCGDN1(enum.Enum):  # Official spec options
+    After15Seconds = 15
+    After30Seconds = 30
+    After1Minute = 60
+    After5Minutes = 300
+    Never = 0
+
+
+class DeviceOffCGDN1(enum.Enum):  # Official spec options
+    After15Minutes = 15
+    After30Minutes = 30
+    After1Hour = 60
+    Never = 0
+
+
+class DisplayTemeratureUnitCGDN1(enum.Enum):
+    Celcius = "c"
+    Fahrenheit = "f"
+
+
+class AirQualityMonitorCGDN1Status:
```

Comment:
```suggestion
class AirQualityMonitorStatus:
```

Reply:
My intention was not to build any generic air quality monitor now, because I'm not sure whether the set of functionalities that `CGDN1` supports are anyhow representative for a wider array of sensors. I think `AirQualityMonitor` can always be created in the future, but establishing it now can result in breaking changes, because other AirQualityMonitorXYZ may support only i.e. `pm 2.5`.

Last time I experienced a similar issue while adding Xiaomi AirPurifier 3C in this PR https://github.com/rytilahti/python-miio/pull/899

Reply:
Fair enough!